include/arch/arc/v2/linker.ld

```diff
@@ -4,9 +4,231 @@
+
+SECTIONS {
+
+#include <linker/rel-sections.ld>
+
+	GROUP_START(ROMABLE_REGION)
+
+	SECTION_PROLOGUE(_TEXT_SECTION_NAME,,ALIGN(1024)) {
+		_image_rom_start = .;
+		_image_text_start = .;
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rom-start.ld>
+		*(.text .text.*)
+		*(.gnu.linkonce.t.*)
+
+	. = ALIGN(4);
+#include <linker/kobject-text.ld>
+	} GROUP_LINK_IN(ROMABLE_REGION)
+
+	_image_text_end = .;
+	_image_rodata_start = .;
+
+#include <linker/common-rom.ld>
+
+	SECTION_PROLOGUE(_RODATA_SECTION_NAME,,) {
+		KEEP(*(.openocd_dbg))
+		KEEP(*(".openocd_dbg.*"))
+		*(".rodata")
+		*(".rodata.*")
+		*(.gnu.linkonce.r.*)
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rodata.ld>
+
+#include <linker/kobject-rom.ld>
+	} GROUP_LINK_IN(ROMABLE_REGION)
+
+#ifdef __MWDT_LINKER_CMD__
+/* TODO: add mwdt specific ROM C++ sections */
+#else
+#include <linker/cplusplus-rom.ld>
+#endif /* __MWDT_LINKER_CMD__ */
+
+	_image_rodata_end = .;
+	MPU_ALIGN(_image_rodata_end - _image_rom_start);
+	_image_rom_end = .;
+	_image_rom_size = _image_rom_end - _image_rom_start;
+
+	GROUP_END(ROMABLE_REGION)
+
+	GROUP_START(RAMABLE_REGION)
+
+#include <app_data_alignment.ld>
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-ram-sections.ld>
+
+#if defined(CONFIG_USERSPACE)
+#define APP_SHARED_ALIGN MPU_MIN_SIZE_ALIGN
+#define SMEM_PARTITION_ALIGN MPU_ALIGN
+
+#include <app_smem.ld>
+
+	_image_ram_start = _app_smem_start;
+	_app_smem_size = _app_smem_end - _app_smem_start;
+	_app_smem_rom_start = LOADADDR(_APP_SMEM_SECTION_NAME);
+#endif /* CONFIG_USERSPACE */
+
+	SECTION_DATA_PROLOGUE(_BSS_SECTION_NAME,(NOLOAD),) {
+		MPU_MIN_SIZE_ALIGN
+		/*
+		 * For performance, BSS section is assumed to be 4 byte aligned and
+		 * a multiple of 4 bytes
+		 */
+		. = ALIGN(4);
+		__bss_start = .;
+		_image_ram_start = .;
+		__kernel_ram_start = .;
+		*(".bss")
+		*(".bss.*")
+		*(COMMON)
+		*(".kernel_bss.*")
+
+		/*
+		 * BSP clears this memory in words only and doesn't clear any
+		 * potential left over bytes.
+		 */
+		__bss_end = ALIGN(4);
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+
+#include <linker/common-noinit.ld>
+
+	SECTION_DATA_PROLOGUE(_DATA_SECTION_NAME,,) {
+
+/* when XIP, .text is in ROM, but vector table must be at start of .data */
+		__data_ram_start = .;
+		*(".data")
+		*(".data.*")
+		*(".kernel.*")
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rwdata.ld>
+
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
+
+	__data_rom_start = LOADADDR(_DATA_SECTION_NAME);
+
+#include <linker/common-ram.ld>
+#include <linker/kobject.ld>
+
+#ifdef __MWDT_LINKER_CMD__
+/* TODO: add mwdt specific RAM C++ sections */
```

Comment:
see above. Also, please add an error message so users get notified C++ isn't supported yet for mwdt.